### PR TITLE
Feature/e UI 4875 unique cases

### PR DESCRIPTION
**Pull request by connorpgpmcelroy**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4875 - Note not completely finished with regards to potential CCD fix for returning all cases

### Change description ###
Sorting by case ID
Adding header to indicate number of unique cases
Fixing pagination issue when filter is applied

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: api/workAllocation2/util.ts [hunk 1]
```diff
@@ -482,7 +488,7 @@ export function getSubstantiveRoles(roleAssignments: RoleAssignment[]): RoleAssi
 }
 
 // Note: array type may need to be changed depending on where pagination called
-export const paginate = (array: Case[], pageNumber: number, pageSize: number): any[] => {
+export const paginate = (array: RoleCaseData[], pageNumber: number, pageSize: number): any[] => {
```
Comment (kuda1992) [inline api/workAllocation2/util.ts:491]: Paginate is a generic function. Can we can this take an `array` generic type. Example code below

`export function paginate<T>(array: T[], pageNumber: number, pageSize: number): T[] {
  return array.slice((pageNumber - 1) * pageSize, pageNumber * pageSize);
};`

File: api/workAllocation2/util.ts [hunk 2]
```diff
@@ -500,3 +506,8 @@ export async function getTypesOfWorkByUserId(path, req: express.Request): Promis
   }
   return null;
 }
+
+export function getUniqueCases(caseData: RoleCaseData[]): number {
```
Comment (kuda1992) [inline api/workAllocation2/util.ts:510]: This should be called `getUniqueCasesCount`

**Review by kuda1992 — CHANGES_REQUESTED** [review]
Few minor changes

File: api/workAllocation2/util.ts (hunk 1 above)
Comment (connorpgpmcelroy) [reply api/workAllocation2/util.ts:491]: good point, will do

File: api/workAllocation2/util.ts (hunk 2 above)
Comment (connorpgpmcelroy) [reply api/workAllocation2/util.ts:510]: sure

**Review by kuda1992 — APPROVED** [review]
Looks good

**Pull request merged**